pyiceberg/io/pyarrow.py

```diff
@@ -689,15 +689,28 @@ def _(obj: pa.StructType, visitor: PyArrowSchemaVisitor[T]) -> T:
     return visitor.struct(obj, results)
 
 
-@visit_pyarrow.register(pa.ListType)
-def _(obj: pa.ListType, visitor: PyArrowSchemaVisitor[T]) -> T:
+def process_list_type(obj: Union[pa.ListType, pa.LargeListType, pa.FixedSizeListType], visitor: PyArrowSchemaVisitor[T]) -> T:
     visitor.before_list_element(obj.value_field)
     result = visit_pyarrow(obj.value_type, visitor)
     visitor.after_list_element(obj.value_field)
-
     return visitor.list(obj, result)
 
 
+@visit_pyarrow.register(pa.ListType)
```

Comment:
I tried to create a new type variable equals to `Union[pa.ListType, pa.LargeListType, pa.FixedSizeListType], visitor: PyArrowSchemaVisitor[T]` and use it in a single register, but MyPy didn't like it:
```
pyiceberg/io/pyarrow.py:695: error: No overload variant of "register" of "_SingleDispatchCallable" matches argument type "<typing special form>"  [call-overload]
pyiceberg/io/pyarrow.py:695: note: Possible overload variants:
pyiceberg/io/pyarrow.py:695: note:     def register(self, cls: type[Any], func: None = ...) -> Callable[[Callable[..., T]], Callable[..., T]]
pyiceberg/io/pyarrow.py:695: note:     def register(self, cls: Callable[..., T], func: None = ...) -> Callable[..., T]
pyiceberg/io/pyarrow.py:695: note:     def register(self, cls: type[Any], func: Callable[..., T]) -> Callable[..., T]
```
So I moved the transformation to a new method with the Union types and I created 3 different register, one for each type of the union.

Reply:
Hi @hussein-awala. I think we can do:
```python
@visit_pyarrow.register(pa.ListType)
@visit_pyarrow.register(pa.LargeListType)
@visit_pyarrow.register(pa.FixedSizeListType)
def _(obj: Union[pa.ListType, pa.LargeListType, pa.FixedSizeListType], visitor: PyArrowSchemaVisitor[T]) -> T:
    visitor.before_list_element(obj.value_field)
    result = visit_pyarrow(obj.value_type, visitor)
    visitor.after_list_element(obj.value_field)
    return visitor.list(obj, result)
```
I checked locally and MyPy did not complain. WDYT?

Reply:
It seems to work, thanks!